cmd/ova-provider-server/ova-provider-server.go

```diff
@@ -514,6 +513,36 @@ func convertToVmStruct(envelope []Envelope, ovaPath []string) ([]VM, error) {
 	return vms, nil
 }
 
+func applyConfiguration(newVM *VM, confs []VirtualConfig) {
+	for _, conf := range confs {
+		switch conf.Key {
+		case "firmware":
+			newVM.Firmware = conf.Value
+		case "memoryHotAddEnabled":
+			newVM.MemoryHotAddEnabled, _ = strconv.ParseBool(conf.Value)
+		case "cpuHotAddEnabled":
+			newVM.CpuHotAddEnabled, _ = strconv.ParseBool(conf.Value)
+		case "cpuHotRemoveEnabled":
+			newVM.CpuHotRemoveEnabled, _ = strconv.ParseBool(conf.Value)
+		}
+	}
+}
+
+func applyExtraConfiguration(newVM *VM, confs []ExtraVirtualConfig) {
```

Comment:
I tried to find the difference between `applyExtraConfiguration` and `applyConfiguration` and I don't see any - if these functions are identical then let's keep `applyConfiguration` and call it twice

Reply:
the only difference is the the object that sent to the func, I initially wanted to use the same object and func since they identical but we need a different xml annotation to get the data correctly.
I though of using an interface for both objects to reduce code duplication but it will be more added code than just this func, so i'm not sure what will be better....

Reply:
I see, ok - I didn't notice the type of the second argument..
I posted another commit to extract the common handling then